pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -15,28 +15,38 @@
 package controller
 
 import (
+	"istio.io/istio/pkg/config/schema/gvr"
+	"istio.io/istio/pkg/kube/controllers"
 	"net"
 	"strconv"
 	"sync"
 
 	"github.com/yl2chen/cidranger"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
 
 	"istio.io/api/label"
 	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/serviceregistry/kube"
 	"istio.io/istio/pkg/cluster"
 	"istio.io/istio/pkg/config/host"
 	"istio.io/istio/pkg/config/mesh"
+	"istio.io/istio/pkg/kube/kclient"
+	"istio.io/istio/pkg/kube/kubetypes"
 	"istio.io/istio/pkg/network"
+	"istio.io/istio/pkg/slices"
 )
 
 type networkManager struct {
 	sync.RWMutex
 	// CIDR ranger based on path-compressed prefix trie
-	ranger              cidranger.Ranger
-	clusterID           cluster.ID
-	meshNetworksWatcher mesh.NetworksWatcher
+	ranger    cidranger.Ranger
+	clusterID cluster.ID
+
+	gatewayResourceClient kclient.Untyped
```

Comment:
It does not seem quite right, we only support watching gateway/other CRD in config cluster.

The proper change should be add a CB handler to CRD client

Reply:
> other CRD in config cluster

Not completely true, we do support watching workloadentry.

Reply:
> The proper change should be add a CB handler to CRD client

Not sure what you mean. According to John, the way we initialize this only takes effect with the CRD present and is robust to the CRDs being added later.

Reply:
same as above comment